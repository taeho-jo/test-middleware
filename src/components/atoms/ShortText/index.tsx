import React from 'react';
// Styles
import { css } from '@emotion/react';
import { commonStyles } from '../../../styles/Common.styles';

// Types
interface PropsType {
  text: string;
  color?: string;
  fontWeight?: string;
  fontSize?: string;
  cursor?: string;
  onClick?: () => void;
}
const ShortText = ({
  text,
  color = commonStyles.colors.b_1,
  fontSize = '16px',
  fontWeight = 'normal',
  cursor = 'default',
  onClick,
}: PropsType) => {
  return (
    <span
      onClick={onClick}
      css={textStyle(color, fontSize, fontWeight, cursor)}
    >
      {text}
    </span>
  );
};

export default ShortText;

const textStyle = (color, fontSize, fontWeight, cursor) => css`
  color: ${color};
  font-size: ${fontSize};
  font-weight: ${fontWeight};
  cursor: ${cursor};
`;
