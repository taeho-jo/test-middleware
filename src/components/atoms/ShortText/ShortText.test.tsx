import React from 'react';
import { render } from '@testing-library/react';
import ShortText from './index';

describe('Short Index', () => {
  it('short text ', () => {
    const utils = render(
      <ShortText
        text={'Short Index'}
        color={'red'}
        fontSize={'20px'}
        fontWeight={'normal'}
        cursor={'default'}
        onClick={() => console.log('console')}
      />,
    );
    expect(utils.container).toMatchSnapshot();
  });

  it('short text ', () => {
    const utils = render(<ShortText text={'Short Index'} color={'red'} />);
    expect(utils.container).toMatchSnapshot();
  });
  it('short text ', () => {
    const utils = render(<ShortText text={'Short Index'} cursor={'pointer'} />);
    expect(utils.container).toMatchSnapshot();
  });
});
