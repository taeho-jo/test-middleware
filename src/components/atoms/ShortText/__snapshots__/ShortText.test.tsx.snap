// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Short Index short text  1`] = `
<div>
  <span
    class="css-1ynmwd2-textStyle"
  >
    Short Index
  </span>
</div>
`;

exports[`Short Index short text  2`] = `
<div>
  <span
    class="css-1c3ti0o-textStyle"
  >
    Short Index
  </span>
</div>
`;

exports[`Short Index short text  3`] = `
<div>
  <span
    class="css-17ewuvp-textStyle"
  >
    Short Index
  </span>
</div>
`;
